restapi.go

```diff
@@ -2552,3 +2562,234 @@ func (s *Session) FollowupMessageEdit(appID string, interaction *Interaction, me
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &st)
+	return
+}
+
+// StartThreadWithoutMessage creates a new thread that is not connected
+// to an existing message.
+// The created thread is always a ChannelTypeGuildPrivateThread
+//
+// POST /channels/{channel.id}/threads
+func (s *Session) StartThreadWithoutMessage(channelID string, data *ThreadCreateData) (st *Channel, err error) {
+	body, err := s.RequestWithBucketID("POST", EndpointThreads(channelID), data, EndpointThreads(""))
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &st)
+	return
+}
+
+// JoinThread joins the session user to a thread
+// PUT/channels/{channel.id}/thread-members/@me
+func (s *Session) JoinThread(channelID string) (err error) {
+	_, err = s.RequestWithBucketID("PUT", EndpointThreadMembers(channelID)+"@me", nil, EndpointThreadMembers(""))
+	return
+}
+
+// AddUserToThread adds another user to a thread
+// PUT /channels/{channel.id}/thread-members/{user.id}
+func (s *Session) AddUserToThread(channelID, uID string) (err error) {
+	_, err = s.RequestWithBucketID("PUT", EndpointThreadMembers(channelID)+uID, nil, EndpointThreadMembers(""))
+	return
+}
+
+// LeaveThread removes the session user from a thread
+// DELETE /channels/{channel.id}/thread-members/@me
+func (s *Session) LeaveThread(channelID string) (err error) {
+	_, err = s.RequestWithBucketID("DELETE", EndpointThreadMembers(channelID)+"@me", nil, EndpointThreadMembers(""))
+	return
+}
+
+// RemoveUserFromThread removes another user from a thread
+// PUT /channels/{channel.id}/thread-members/{user.id}
+func (s *Session) RemoveUserFromThread(channelID, uID string) (err error) {
+	_, err = s.RequestWithBucketID("DELETE", EndpointThreadMembers(channelID)+uID, nil, EndpointThreadMembers(""))
+	return
+}
+
+// ThreadMembers lists the members of a thread
+// GET /channels/{channel.id}/threads-members
+func (s *Session) ThreadMembers(channelID string) (st []*ThreadMember, err error) {
+	body, err := s.RequestWithBucketID("GET", EndpointThreadMembers(channelID), nil, EndpointThreadMembers(""))
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &st)
+	return
+}
+
+// ThreadsResponseBody is the returned data from the discord gateway
+// of some specific threads requests
+type ThreadsResponseBody struct {
+	Threads []*Channel      `json:"threads"`
+	Members []*ThreadMember `json:"members"`
+
+	// Ehether there are potentially additional threads
+	// that could be returned on a subsequent call
+	HasMore bool `json:"has_more"`
+}
+
+// ActiveThreads returns all active threads in the channel
+// GET /channels/{channel.id}/threads/active
+func (s *Session) ActiveThreads(channelID string) (st *ThreadsResponseBody, err error) {
+	body, err := s.RequestWithBucketID("GET", EndpointChannelActiveThreads(channelID), nil, EndpointThreadMembers(""))
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &st)
+	return
+}
```

Comment:
It looks like this endpoint [is already deprecated](https://discord.com/developers/docs/resources/channel#list-active-threads). We might just want to move to the guild active threads endpoint to begin with; that one also has a more reasonable response structure, such that we could return a slice of threads and a slice of members rather than a custom struct specifically for one function.